### PR TITLE
chore(ci): fix flakiness, misc improvements

### PR DESCRIPTION
**Pull request by anshbansal**
`venv` contains `python`. Caching whole `venv` instead of just `pip` cache means you are backing up python which is problematic and is probably causing python not being found randomly in the smoke tests. If it is causing issues with smoke tests it would also cause issues with `metadata-ingestion` so removing that too.

e.g. https://github.com/datahub-project/datahub/actions/runs/4433850334/jobs/7779664856 where log says
```
+++ dirname ./smoke-test/run-quickstart.sh
++ cd ./smoke-test
++ pwd
+ DIR=/home/runner/work/datahub/datahub/smoke-test
+ cd /home/runner/work/datahub/datahub/smoke-test
+ python3 -m venv venv
Error: [Errno 2] No such file or directory: '/home/runner/work/datahub/datahub/smoke-test/venv/bin/python3': '/home/runner/work/datahub/datahub/smoke-test/venv/bin/python3'
```
Python not being found is really odd as we have a setup python step. It is probably the `venv` cache after that which is messing things somehow.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: .github/workflows/metadata-ingestion.yml [hunk 1]
```diff
@@ -54,26 +54,10 @@ jobs:
...
-      - name: print dependencies
-        id: print-dependency
-        run: cat /tmp/would_be_installed.txt
-      - uses: actions/cache@v3
-        id: cache-venv
-        with:
-          path: ./metadata-ingestion/venv/
-          key: ${{ runner.os }}-venv-${{ steps.packages_checksum.outputs.packages_checksum }}-${{ matrix.python-version }}
+      - name: Install package
+        run: ./gradlew :metadata-ingestion:installPackageOnly
```
Comment (hsheth2) [inline .github/workflows/metadata-ingestion.yml:58]: Is this just for debugging?

File: metadata-ingestion/build.gradle [hunk 2]
```diff
@@ -12,7 +12,8 @@ if (!project.hasProperty("extra_pip_requirements")) {
 }
 
 task checkPythonVersion(type: Exec) {
-  commandLine python_executable, '-c', 'import sys; assert sys.version_info >= (3, 7)'
+  commandLine python_executable, '-c',
+    'import sys; assert (3, 11) > sys.version_info >= (3, 7), f"Python version {sys.version_info[:2]} not allowed"'
```
Comment (hsheth2) [inline metadata-ingestion/build.gradle:16]: Change message to "Is too old"

**Review by hsheth2 — APPROVED** [review]
Unsure if we want to remove the venv cache wholesale vs just commenting out

**Comment by hsheth2** [comment]
Also it seems strange that we've seen issues in smoke test but not with metadata ingestion yet. Any ideas why that could be?

File: .github/workflows/docker-unified.yml [hunk 3]
```diff
@@ -490,20 +490,6 @@ jobs:
...
-
-          # only the last line of the output is the packages that will be installed
-          pip install --dry-run -r ./smoke-test/requirements.txt  | tail -n 1 > /tmp/would_be_installed.txt
-          CHECKSUM=$(shasum /tmp/would_be_installed.txt | awk '{print $1}')
-          echo "packages_checksum=$CHECKSUM" >> $GITHUB_OUTPUT
-      - uses: actions/cache@v3
-        id: cache-venv
-        with:
-          path: ./smoke-test/venv/
-          key: smoke-venv-${{ runner.os }}-${{ steps.packages_checksum.outputs.packages_checksum }}
```
[the remark this answers is not in the document]
Comment (anshbansal) [reply .github/workflows/docker-unified.yml:506]: There is no difference in run time. See `master` and this branch's runs. No time difference is there.

File: .github/workflows/metadata-ingestion.yml (hunk 1 above)
Comment (anshbansal) [reply .github/workflows/metadata-ingestion.yml:58]: No. Wanted to see how much time installing the package takes separately to the tests.

File: metadata-ingestion/build.gradle (hunk 2 above)
Comment (anshbansal) [reply metadata-ingestion/build.gradle:16]: What do you mean? If someone is using 3.11 it won't be too old but too new.

File: .github/workflows/docker-unified.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (hsheth2) [reply .github/workflows/docker-unified.yml:506]: Weird - that probably means the caching wasn't even working?

File: metadata-ingestion/build.gradle (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/build.gradle:16]: oh i missed the 3.11 part - then this is fine

**Pull request merged**